diff --git a/udev b/udev
index 4898b89..5645e45 100644
--- a/udev
+++ b/udev
@@ -45,6 +45,7 @@ udev_run() {
 	sh -c "exec 4< /dev/console" || { exec 0> /dev/null; exec 1> /dev/null; exec 2> /dev/null; }
 
 	$_UDEV_DAEMON --daemon
-	udevadm trigger --action=add
+	udevadm trigger --action=add --subsystem-match=block \
+			--subsystem-match=graphics --subsystem-match=drm --subsystem-match=tty
 	udevadm settle
 }
