diff --git a/udev b/udev
index 87551ff4a9..d8debf3b53 100644
--- a/udev
+++ b/udev
@@ -36,6 +36,9 @@ udev_enabled() {
 }
 
 udev_run() {
+	# Workaround if console=null for systemd-udevd which needs a valid stdin, stdout and stderr to work
+	sh -c "exec 4< /dev/console" || { exec 0> /dev/null; exec 1> /dev/null; exec 2> /dev/null; }
+
 	add_module_pre_hook "udev_shutdown_hook_handler"
 
 	mkdir -p /run
