From 4c809c0be348d1ce53540cf3a2b5cc71fc87c436 Mon Sep 17 00:00:00 2001
From: Sergio Prado <sergio.prado@toradex.com>
Date: Wed, 8 Feb 2023 12:52:09 +0000
Subject: [PATCH] Add targets to Makefile

Signed-off-by: Sergio Prado <sergio.prado@toradex.com>
---
 Makefile | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/Makefile b/Makefile
index c7e507293ef0..fd389ff2b79a 100644
--- a/Makefile
+++ b/Makefile
@@ -3,3 +3,9 @@
 obj-m += composefs.o
 
 composefs-objs += cfs-reader.o cfs.o
+
+modules:
+	make -C ${KERNEL_SRC} M=${PWD} modules
+
+modules_install:
+	make -C ${KERNEL_SRC} M=${PWD} modules_install
+
+clean:
+	make -C ${KERNEL_SRC} M=${PWD} clean
-- 
2.25.1

