From 51e9d1cb0f6807c84aaf4da76728254aef6c900d Mon Sep 17 00:00:00 2001
From: Sergio Prado <sergio.prado@toradex.com>
Date: Thu, 30 Mar 2023 16:17:13 -0300
Subject: [PATCH] ostree-convert-commit.py: calculate fsverity digest

Signed-off-by: Sergio Prado <sergio.prado@toradex.com>
---
 tools/ostree-convert-commit.py | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/tools/ostree-convert-commit.py b/tools/ostree-convert-commit.py
index e52eadc5f533..d132b88b6f74 100755
--- a/tools/ostree-convert-commit.py
+++ b/tools/ostree-convert-commit.py
@@ -21,6 +21,7 @@ import sys
 import stat
 import gi
 import base64
+import subprocess
 
 gi.require_version('OSTree', '1.0')
 from gi.repository import Gio, GLib, OSTree
@@ -71,6 +72,9 @@ def recurse_dir(repo, parent, d):
         if i.get_file_type() == Gio.FileType.REGULAR:
             csum = f.get_checksum()
             entry["x-payload"] = get_payload(csum)
+            file = sys.argv[1] + "/objects/" + get_payload(csum)
+            digest = subprocess.run(["fsverity", "digest", "--compact", file], stdout=subprocess.PIPE)
+            entry["digest"] = "sha256:" + digest.stdout.decode("utf-8").rstrip('\n')
         elif i.get_file_type() == Gio.FileType.SYMBOLIC_LINK:
             entry["linkName"] = i.get_attribute_byte_string("standard::symlink-target")
 
-- 
2.25.1

