Submodule aktualizr contains modified content
diff --git a/aktualizr/src/libaktualizr/CMakeLists.txt b/aktualizr/src/libaktualizr/CMakeLists.txt
index d868eaf6d..986e6c1f9 100644
--- a/aktualizr/src/libaktualizr/CMakeLists.txt
+++ b/aktualizr/src/libaktualizr/CMakeLists.txt
@@ -73,7 +73,7 @@ add_library(aktualizr_lib SHARED
 target_link_libraries(aktualizr_lib ${AKTUALIZR_EXTERNAL_LIBS})
 set_target_properties(aktualizr_lib PROPERTIES LIBRARY_OUTPUT_NAME aktualizr)
 install(TARGETS aktualizr_lib LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT aktualizr)
-install(DIRECTORY ${CMAKE_SOURCE_DIR}/include/libaktualizr DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/../../include/libaktualizr DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
 install(FILES ${CMAKE_CURRENT_BINARY_DIR}/../../jsoncpp/json/json.h DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/libaktualizr/json)
 
 configure_file(aktualizr.pc.in aktualizr.pc @ONLY)
