Upstream-Status: Pending

Index: git/test/helpers.bash
===================================================================
--- git.orig/test/helpers.bash
+++ git/test/helpers.bash
@@ -575,7 +575,7 @@
     fi
 
     nsenter -n -t "${CONTAINER_NS_PIDS[$container_ns]}" timeout --foreground -v --kill=10 5 \
-        nc $nc_common_args -l -p $container_port &>"$NETAVARK_TMPDIR/nc-out" <$stdin &
+        ncat $nc_common_args -l -p $container_port &>"$NETAVARK_TMPDIR/nc-out" <$stdin &
 
     # make sure to wait until port is bound otherwise test can flake
     # https://github.com/containers/netavark/issues/433
@@ -588,7 +588,7 @@
     fi
 
     data=$(random_string)
-    run_in_host_netns nc $nc_common_args $connect_ip $host_port <<<"$data"
+    run_in_host_netns ncat $nc_common_args $connect_ip $host_port <<<"$data"
 
     got=$(cat "$NETAVARK_TMPDIR/nc-out")
     assert "$got" == "$data" "ncat received data"
