From e7507f802575c4f9b81007e270fced2abcfbd5b0 Mon Sep 17 00:00:00 2001
From: Sergio Prado <sergio.prado@toradex.com>
Date: Thu, 28 Sep 2023 10:32:46 +0000
Subject: [PATCH] Add TorizonCore specific groups

Upstream-Status: Inappropriate [configuration]

Signed-off-by: Eduardo Ferreira <eduardo.barbosa@toradex.com>
Signed-off-by: Sergio Prado <sergio.prado@toradex.com>
Signed-off-by: Stefan Agner <stefan.agner@toradex.com>
---
 group.master | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/group.master b/group.master
index 5b622841be8f..f115390feecb 100644
--- a/group.master
+++ b/group.master
@@ -35,9 +35,15 @@
 plugdev:*:46:
 kvm:*:47:
 sgx:*:48:
+gpio:*:49:
 staff:*:50:
+i2cdev:*:51:
+spidev:*:52:
+pwm:*:54:
 games:*:60:
 shutdown:*:70:
 wheel:*:80:
+nobody:*:99:
 users:*:100:
+render:*:103:
 nogroup:*:65534:
-- 
2.25.1

