From 4f375db7f361f482e11195d7f705745477686bd5 Mon Sep 17 00:00:00 2001
From: andreriesco <andre.riesco@toradex.com>
Date: Tue, 4 May 2021 09:40:38 -0300
Subject: [PATCH] Add TorizonCore OS

---
 neofetch | 39 +++++++++++++++++++++++++++++++++++++--
 1 file changed, 37 insertions(+), 2 deletions(-)

diff --git a/neofetch b/neofetch
index 1e4b564..b453850 100755
--- a/neofetch
+++ b/neofetch
@@ -796,7 +796,7 @@ image_source="auto"
 #       Regata, Rosa, sabotage, Sabayon, Sailfish, SalentOS, Scientific,
 #       Septor, SereneLinux, SharkLinux, Siduction, Slackware, SliTaz,
 #       SmartOS, Solus, Source_Mage, Sparky, Star, SteamOS, SunOS,
-#       openSUSE_Leap, openSUSE_Tumbleweed, openSUSE, SwagArch, Tails,
+#       openSUSE_Leap, openSUSE_Tumbleweed, openSUSE, SwagArch, Tails, TorizonCore
 #       Trisquel, Ubuntu-Budgie, Ubuntu-GNOME, Ubuntu-MATE, Ubuntu-Studio,
 #       Ubuntu, Venom, Void, Obarun, windows10, Windows7, Xubuntu, Zorin,
 #       and IRIX have ascii logos
@@ -4938,7 +4938,7 @@ ASCII:
                                 Regata, Rosa, sabotage, Sabayon, Sailfish, SalentOS, Scientific,
                                 Septor, SereneLinux, SharkLinux, Siduction, Slackware, SliTaz,
                                 SmartOS, Solus, Source_Mage, Sparky, Star, SteamOS, SunOS,
-                                openSUSE_Leap, openSUSE_Tumbleweed, openSUSE, SwagArch, Tails,
+                                openSUSE_Leap, openSUSE_Tumbleweed, openSUSE, SwagArch, Tails, TorizonCore
                                 Trisquel, Ubuntu-Budgie, Ubuntu-GNOME, Ubuntu-MATE, Ubuntu-Studio,
                                 Ubuntu, Venom, Void, Obarun, windows10, Windows7, Xubuntu, Zorin,
                                 and IRIX have ascii logos
@@ -9937,6 +9937,41 @@ Nsyh+-..+y+-   yMMMMd   :mMM+
 EOF
         ;;
 
+        "TorizonCore"*)
+            set_colors fg 3 4 8 5
+            read -rd '' ascii_data <<'EOF'
+${c2}                          `.::-.
+                       `-://////:-.
+                     `:////////////:.
+                `.-.`  `-:///////-.     `
+             `.://///:.`  `-::-.     `-://:.`
+           .:///////////:.        `-:////////:.`
+           `.://///////:-`       .:////////////:`
+    `.-/:-`   `.:///:-`    ${c3}`-+o/.${c2}  `.://////-.     `.`
+ `.:///////:-`   `.`    ${c3}`-+sssssso/.${c2}  `.--.     `-:///:-`
+-/////////////:       ${c3}`+sssssssssssso-${c2}       `-://///////:-`
+  .-///////:.`    ``    ${c3}-/sssssssso/.${c2}    `   .:///////////-.
+     .-::.`    `-://:-.    ${c3}-/sso/.${c2}    `.:/:-.  `.://///-.
+            `-:////////:-.    ${c3}`${c2}    `.:////////-.  `.-.
+${c1}o-${c2}          -:///////////:`       .:////////////:`        ${c1}-o${c2}
+${c1}hdho-${c2}         `-://///:.`    `..`   `-://////:.`       ${c1}-ohdh${c2}
+${c1}/ydddho-${c2}         `--.`    `.:////:.`   `-::.`       ${c1}-ohdddy/${c2}
+  ${c1}./ydddho:`${c2}           `.://////////:.          ${c1}`:ohdddy/.${c2}
+`    ${c1}`/shddho:`${c2}        `.://////////:.       ${c1}`:ohddds/`${c2}    `
+::-`    ${c1}`/shddhs:`${c2}        `.:////:.`      ${c1}`:shddhs/`${c2}    `-::
+:///:-`    ${c1}`:shddhs/`${c2}        `..`      ${c1}`/shddhs:`${c2}    `-:///-
+ `.:///:-`    ${c1}`:ohddhs/`${c2}            ${c1}`/shddho:`${c2}    `-:///:.`
+    `.:///:-.     ${c1}-ohdddy/.${c2}      ${c1}./ydddho-${c2}     .-:///-.`
+       `.:///:-.     ${c1}-+hdddy+//+ydddh+-${c2}     .-://:-.
+          `.-///:-.     ${c1}-+yddddddy+-${c2}     .-://:-.
+              .-://:-.     ${c1}./++/.${c2}     .-///:-.
+                 .-:///:.`        `.:///:-`
+                    `-:///:.````.:///:-`
+                       `-:////////:-`
+                          `-::::-`
+EOF
+        ;;
+
         "Trisquel"*)
             set_colors 4 6
             read -rd '' ascii_data <<'EOF'
-- 
2.17.1

