From 77e1f5dea20f75a74df53ac7c23e184612865cf4 Mon Sep 17 00:00:00 2001
From: Sergio Prado <sergio.prado@toradex.com>
Date: Tue, 24 May 2022 10:30:46 -0300
Subject: [PATCH] cli: use go cross compiler

Signed-off-by: Sergio Prado <sergio.prado@toradex.com>
Signed-off-by: Bruce Ashfield <bruce.ashfield@gmail.com>
---
 cli/scripts/build/binary | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cli/scripts/build/binary b/cli/scripts/build/binary
index e4c5e12a6b44..c5151c9984ee 100755
--- a/cli/scripts/build/binary
+++ b/cli/scripts/build/binary
@@ -75,6 +75,6 @@ echo "Building $GO_LINKMODE $(basename "${TARGET}")"
 
 export GO111MODULE=auto
 
-go build -o "${TARGET}" -tags "${GO_BUILDTAGS}" --ldflags "${LDFLAGS}" ${GO_BUILDMODE} "${SOURCE}"
+${GO} build -o "${TARGET}" -tags "${GO_BUILDTAGS}" --ldflags "${LDFLAGS}" ${GO_BUILDMODE} "${SOURCE}"
 
 ln -sf "$(basename "${TARGET}")" "$(dirname "${TARGET}")/docker"
-- 
2.25.1

